engine.js

```diff
@@ -0,0 +1,96 @@
+"format cjs";
+
+var wrap = require('word-wrap');
+var cloneDeep = require('lodash.clonedeep');
+var map = require('lodash.map');
+var longest = require('longest');
+var rightPad = require('right-pad');
+
+// This can be any kind of SystemJS compatible module.
+// We use Commonjs here, but ES6 or AMD would do just
+// fine.
+module.exports = function (options) {
+
+  options = cloneDeep(options || {});
+  var types = options.types;
+
+  var length = longest(Object.keys(types)).length + 1;
+  var choices = map(types, function (type, key) {
```

Comment:
@LinusU Why prefer an array? Pros I see include ability to dictate order, if that is a significant concern here. Cons are it being less easy to modify - using array searches and filters rather than direct manipulation by key, unless converting first to object.